ast/desugar/Desugar.cc

```diff
@@ -37,6 +37,43 @@ core::NameRef blockArg2Name(DesugarContext dctx, const BlockArg &blkArg) {
     return blkIdent->name;
 }
 
+// Get the num from the name of the Node if it's a LVar.
+// Return -1 otherwise.
+int numparamNum(DesugarContext dctx, parser::Node *decl) {
+    if (auto *lvar = parser::cast_node<parser::LVar>(decl)) {
+        auto name_str = lvar->name.show(dctx.ctx);
+        return name_str[1] - 48;
+    }
+    return -1;
+}
+
+// Get the highest numparams used in `decls`
+// Return 0 if the list of declarations is empty.
+int numparamMax(DesugarContext dctx, parser::NodeVec *decls) {
+    int max = 0;
+    for (auto &decl : *decls) {
+        auto num = numparamNum(dctx, decl.get());
+        if (num > max) {
+            max = num;
+        }
+    }
+    return max;
+}
+
+// Create a local variable from the first declaration for the name "_num" from all the `decls` if any.
+// Return a dummy variable if no declaration is found for `num`.
+TreePtr numparamTree(DesugarContext dctx, int num, parser::NodeVec *decls) {
+    for (auto &decl : *decls) {
+        if (auto *lvar = parser::cast_node<parser::LVar>(decl.get())) {
+            if (numparamNum(dctx, decl.get()) == num) {
+                return MK::Local(lvar->loc, lvar->name);
+            }
+        }
```

Comment:
Should this be an ENFORCE if this cast ever fails?

Reply:
(i.e., it's fine for it to fallback to the dummy `MK::Local` if this search fails to find something so that sorbet doesn't crash, but would that be an expected situation or a bug in sorbet?)

Reply:
oh i see, we might be making locals for like `_1` and `_2` only because we saw `_3` in the body. makes sense, i see that the comment is getting at that now.